stratum/hal/bin/barefoot/README.md

```diff
@@ -5,345 +5,36 @@ Copyright 2018-present Open Networking Foundation
-    -bf_pipeline_config_binary_file=$PWD/device_config.pb.bin \
-    -unpack_dir=$PWD
-```
-
-### bf-p4c Archive Format
-
-You can enable a special archive format by passing the
-`-incompatible_enable_p4_device_config_tar` flag when starting Stratum.
-
-*Note: Support for this format may disappear at any time.*
-
-The format accepts a tar archive of the bf-p4c compiler output:
-
-```bash
-mkdir -p /tmp/p4out
-bf-p4c <options...> \
-  -o /tmp/p4out -I ${P4_SRC_DIR} \
-  --p4runtime-files /tmp/p4out/p4info.txt \
-  --p4runtime-force-std-externs \
-  ${P4_SRC_DIR}/my_prog.p4
-tar -czf "pipeline.tgz" -C "/tmp/p4out" .
-```
-
-The structure of the tar archive should look something like this:
-
-```bash
-$ tar -tf pipeline.tgz
-./
-./bfrt.json
-./my_prog.conf
-./pipe/
-./pipe/context.json
-./pipe/tofino.bin
-```
-
-To use this format, enable it with the aforementioned flag when starting
-Stratum, and then use the contents of the tar archive as the `p4_device_config`
-field of the P4Runtime `ForwardingPipelineConfig` message.
-
-## Using p4runtime-shell
-
-[p4runtime-shell](https://github.com/p4lang/p4runtime-shell) is an interactive
-Python shell for P4Runtime. While it can also be used to set the P4 forwarding
-pipeline and issue P4Runtime `Write` RPCs, it especially comes in handy when you
-want to read the forwarding state of the switch.
-
-To start a shell session, you can use (requires Docker):
-```
-./p4runtime-sh-docker --grpc-addr <Stratum IP>:9559 --device-id 1 --election-id 0,1
-```
-
-Refer to the [p4runtime-shell](https://github.com/p4lang/p4runtime-shell)
-documentation for more information.
-
-## Troubleshooting
-
-### Huge pages / DMA allocation error
-
-`ERROR: bf_sys_dma_buffer_alloc for dev_id 0 failed(-1)`
-
-This error means that the Tofino driver could not allocate DMA memory from the
-huge pages pool. Ensure that at least 128 huge pages are mounted and available:
-
-```bash
-> grep HugePages_ /proc/meminfo
-HugePages_Total:     128
-HugePages_Free:      128
-HugePages_Rsvd:        0
-HugePages_Surp:        0
-```
-
-To enable them or allocate more, follow the steps from the [post install script](deb/postinst).
-
-### ONLP / BMC errors on Wedge100BF
-
-`07-24 23:10:16.072010 [x86_64_accton_wedge100bf_32x] ERROR: bmc_send_command(cat /sys/bus/i2c/drivers/com_e_driver/4-0033/temp2_input
-) timed out`
-
-`07-25 08:30:59.834213 [x86_64_accton_wedge100bf_32x] Unable to read status from file (/sys/bus/i2c/drivers/lm75/3-0048/temp1_input)`
-
-This error occurs when ONLP can not reach the BMC CPU managing the platform.
-Either the ONL image is not correctly set up or ONLP support is simply broken on
-this particular switch model. As a workaround, [BSP mode](#Running-with-BSP-or-on-Tofino-model),
-which bypasses ONLP, is available.
+**[Testing Stratum on a Barefoot Tofino based switch](./README.test.md)**
+provides some examples of how to interact with Stratum running on a
+Barefoot Tofino based switch.
```

Comment:
Should we reorder this to Building -> Running -> device config -> testing?

Reply:
Yes, that probably makes sense logically.